espnet/nets/pytorch_backend/conformer/encoder_layer.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Johns Hopkins University (Shinji Watanabe)
+#                Northwestern Polytechnical University (Pengcheng Guo)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Encoder self-attention layer definition."""
+
+import torch
+
+from torch import nn
+
+from espnet.nets.pytorch_backend.transformer.layer_norm import LayerNorm
+
+
+class EncoderLayer(nn.Module):
+    """Encoder layer module.
+
+    :param int size: input dim
+    :param espnet.nets.pytorch_backend.conformer.attention.
+        MultiHeadedAttention self_attn: self attention module
+    :param espnet.nets.pytorch_backend.conformer.positionwise_feed_forward.
+        PositionwiseFeedForward feed_forward:
+        feed forward module
+    :param espnet.nets.pytorch_backend.conformer.positionwise_feed_forward.
+        PositionwiseFeedForward feed_forward:
+        feed forward module
+    :param espnet.nets.pytorch_backend.conformer.convolution.
+        ConvolutionBlock feed_foreard:
+        feed forward module
+    :param float dropout_rate: dropout rate
+    :param bool normalize_before: whether to use layer_norm before the first block
+    :param bool concat_after: whether to concat attention layer's input and output
+        if True, additional linear will be applied.
+        i.e. x -> x + linear(concat(x, att(x)))
+        if False, no additional linear will be applied. i.e. x -> x + att(x)
+    :param bool rel_pos: whether to use relative positional encoding
+    :param bool macaron_style: whether to use macaron style for PositionwiseFeedForward
+
+    """
+
+    def __init__(
+        self,
+        size,
+        self_attn,
+        feed_forward,
+        feed_forward2,
+        conv_block,
+        dropout_rate,
+        normalize_before=True,
+        concat_after=False,
+        rel_pos=False,
+        macaron_style=False,
+    ):
+        """Construct an EncoderLayer object."""
+        super(EncoderLayer, self).__init__()
+        self.self_attn = self_attn
+        self.feed_forward = feed_forward
+        self.feed_forward2 = feed_forward2
+        self.norm1 = LayerNorm(size)  # for the first FNN module
+        self.norm2 = LayerNorm(size)  # for MHA module
+        self.norm3 = LayerNorm(size)  # for CNN module
+        self.norm4 = LayerNorm(size)  # for the second FNN module
+        self.norm5 = LayerNorm(size)
+        self.conv_block = conv_block
```

Comment:
Instead of norm1 ... 5 with comments, I prefer
```python
self.norm_ff1 = LayerNorm(size)
self.norm_ff2 = LayerNorm(size)
self.norm_mha = LayerNorm(size)
if self.conv_block is not None:
     self.norm_cnn1 = LayerNorm(size)
     self.norm_cnn2 = LayerNorm(size)
```

Reply:
Thanks so much. I think it's better to use `self.norm_ff1` etc. I will modify these name.